paimon-core/src/main/java/org/apache/paimon/table/system/StatisticTable.java

```diff
@@ -121,15 +127,26 @@ public Table copy(Map<String, String> dynamicOptions) {
 
     private class StatisticScan extends ReadOnceTableScan {
 
+        private @Nullable LeafPredicate snapshotIdPredicate;
+
         @Override
         public InnerTableScan withFilter(Predicate predicate) {
-            // TODO
+            if (predicate == null) {
+                return this;
+            }
+
+            Map<String, LeafPredicate> leafPredicates =
+                    predicate.visit(LeafPredicateExtractor.INSTANCE);
+            snapshotIdPredicate = leafPredicates.get("snapshot_id");
```

Comment:
SNAPSHOT_ID

Reply:
also in line 110

Reply:
TABLE_TYPE maybe not need set with constant value，keep the style with other table seems better，other position would change to SNAPSHOT_ID.